src/util/range.h

```diff
@@ -337,6 +419,18 @@ struct ranget final
       concat_begin, concat_end);
   }
 
+  template <typename other_iteratort>
+  ranget<zip_iteratort<iteratort, other_iteratort>>
+  zip(ranget<other_iteratort> other)
```

Comment:
It might be nice to have an overload which applies `make_range` and calls the existing overload in the case where `other` is an STL container instead of a range. That way your example could be -
`make_range(components).zip(rhs.operands())` instead of `make_range(components).zip(make_range(rhs.operands()))`.